### DISCUSSION TITLE
How to view and edit iModel project extents

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#5005] [Q&A] [answered]
Hello - as part of the work developing the FutureOn FieldTwin to iTwin connector, I would like to be able to view the project extents of an iModel, and if necessary be able to change the project extents. In particular, the maximum depth below sea-level of the iModel. 

The reason for this request is that some elements of the FieldTwin subsea model are not being converted, and I think it may be because they are below the maximum depth of the iModel project extents, so it would be helpful to be able to view the current settings and if necessary change the maximum depth below sea-level.

I think there used to be a way to view and change the iModel project extents, but I think the iModel UI may have changed recently.

**Comment by pmconne** [comment, marked as answer]
The editor packages contain a [set of tools](https://github.com/iTwin/itwinjs-core/blob/ee968025e3be9e367f97289e48e33a24b8c0af6e/editor/frontend/src/ProjectLocation/ProjectExtentsDecoration.ts#L945) that can display and/or modify the project extents and location.

**Reply by PeterJBassett** [reply]
Hello @pmconne - thanks for your reply. So is it only possible to view or change the extents using the typescript API, and is not possible to do via the iModel user interface?

Also, is it possible to view the project extents via an sql query using the iModel Console.

**Reply by pmconne** [reply]
The project extents are stored in the be_Props table:
![image](https://user-images.githubusercontent.com/22944042/215755238-8797f277-e5dd-4dd6-87e3-7bbaa490231e.png)
As far as I'm aware, you can't query the be_Props table using ECSql. You can query it using ordinary SQL.

By "iModel iuser interface" are you referring to iModelHub, or something else (what)?
As far as I'm aware, the iModelHub user interface does not provide a UI for interactively modifying the 3d project extents. It (used to?) provide an API to define a 2d bounding rectangle on a map, which would not help you with adjusting the depth.

If you want to change the contents of the iModel, including the extents, use the TypeScript API.

**Reply by PeterJBassett** [reply]
Thanks again @pmconne for all the information, that is very helpful.
